js-jasmine/src/gilded_rose.js

```diff
@@ -7,59 +7,70 @@ class Item {
 }
 
 class Shop {
-  constructor(items=[]){
-    this.items = items;
+  constructor(items = []) {
+    this.items= items; //instantiates this.items[i]s array
+  }
+
+  removeQuality() {
+    return this.items[i].quality -= this.items[i].quality
+  }
+  decreaseSellIn() {
+    return this.items[i].sellIn -= 1
+  }
+  increaseQuality() {
+    return this.items[i].quality += 1; 
+  }
+  decreaseQuality() {
+   return this.items[i].quality -= 1
+  }
+  isAgedBrie() {
+     this.name == 'Aged Brie'
+  }
+   isBackstagePass() {
+     this.name == 'Backstage passes to a TAFKAL80ETC concert'
+  }
+   isSalfuras() {
+     this.name == 'Sulfuras, Hand of Ragnaros'
   }
   updateQuality() {
-    for (var i = 0; i < this.items.length; i++) {
-      if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-        if (this.items[i].quality > 0) {
-          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-            this.items[i].quality = this.items[i].quality - 1;
+    for (let i = 0; i < this.items.length; i++) {
+      try {
+        //The Quality of an item is never negative & never more than 50
+        if (this.items[i].quality > 0 && this.items[i].quality < 50) {
+          this.decreaseSellIn()
+
+          if (this.isSalfuras()) {
+            this.increaseQuality();
           }
-        }
```

Comment:
Well done for breaking up the code! I especially like how you created separate methods for also updating selling date, quality and checking for the type of product, so you only need to refer back to these methods later, instead of writing them out again and again

Reply:
👼🏿